db_app.py

```diff
@@ -606,35 +632,45 @@ def jobs(job_id):
+        except ValueError as e:
+            raise HTTPRequestError(str(e), 404)
+        return jsonify(job_results)
     else:
-        raise ValueError("Format " + str(fmt) + " not known")
+        raise HTTPRequestError("Format " + str(fmt) + " not known")
 
 
 def failure(job_id):
     logger.info("Retrieving failure for job with ID " + str(job_id))
-    failure = get_hive().get_job_failure_msg_by_id(job_id)
+    try:
+        failure = get_hive().get_job_failure_msg_by_id(job_id)
+    except ValueError as e:
+        raise HTTPRequestError(str(e), 404)
     return jsonify({"msg": failure.msg})
 
 
 def job_email(email, job_id):
     logger.info("Retrieving job with ID " + str(job_id) + " for " + str(email))
-    job = get_hive().get_job_by_id(job_id)
-    results = get_hive().get_result_for_job_id(job_id)
-    if results['status'] == 'complete':
-        results['subject'] = 'Copy database from %s to %s successful' % (
-        results['output']['source_db_uri'], results['output']['target_db_uri'])
-        results['body'] = 'Copy from %s to %s is successful\n' % (
-        results['output']['source_db_uri'], results['output']['target_db_uri'])
-        results['body'] += 'Copy took %s' % (results['output']['runtime'])
-    elif results['status'] == 'failed':
-        failure = get_hive().get_job_failure_msg_by_id(job_id)
-        results['subject'] = 'Copy database from %s to %s failed' % (
-        results['input']['source_db_uri'], results['input']['target_db_uri'])
-        results['body'] = 'Copy failed with following message:\n'
-        results['body'] += '%s\n\n' % (failure.msg)
-        results['body'] += 'Please see URL for more details: %s%s \n' % (results['input']['result_url'], job_id)
-    results['output'] = None
+    try:
+        job = get_hive().get_job_by_id(job_id)
+        results = get_hive().get_result_for_job_id(job_id)
+        if results['status'] == 'complete':
+            results['subject'] = 'Copy database from %s to %s successful' % (
+            results['output']['source_db_uri'], results['output']['target_db_uri'])
+            results['body'] = 'Copy from %s to %s is successful\n' % (
+            results['output']['source_db_uri'], results['output']['target_db_uri'])
+            results['body'] += 'Copy took %s' % (results['output']['runtime'])
+        elif results['status'] == 'failed':
+            failure = get_hive().get_job_failure_msg_by_id(job_id)
+            results['subject'] = 'Copy database from %s to %s failed' % (
+            results['input']['source_db_uri'], results['input']['target_db_uri'])
+            results['body'] = 'Copy failed with following message:\n'
+            results['body'] += '%s\n\n' % (failure.msg)
+            results['body'] += 'Please see URL for more details: %s%s \n' % (results['input']['result_url'], job_id)
+        results['output'] = None
+    except ValueError as e:
+        raise HTTPRequestError(str(e), 404)
```

Comment:
IMHO I would have return a 400 if job_id is not retrieved or email malformed, and a 500 otherwise. I guess this woudl be done while refactoring the code anyway